ml-agents/mlagents/utils/push_to_hf.py

```diff
@@ -0,0 +1,182 @@
+        # Get trainer_type
+        model_name = data["behaviors"][env_id]["trainer_type"]
+
+    # Step 2: Create the metadata
+    metadata = _generate_metadata(model_name, env_id)
+
+    # Step 3: Generate the model card
+    model_card = f"""
+  # **{model_name}** Agent playing **{env_id}**
+  This is a trained model of a **{model_name}** agent playing **{env_id}** using the [Unity ML-Agents Library](https://github.com/Unity-Technologies/ml-agents).
+  """
+
+    model_card += f"""
+  ## Usage (with ML-Agents)
+  The Documentation: https://unity-technologies.github.io/ml-agents/ML-Agents-Toolkit-Documentation/
+  We wrote a complete tutorial to learn to train your first agent using ML-Agents and publish it to the Hub:
+  - A *short tutorial* where you teach Huggy the Dog 🐶 to fetch the stick and then play with him directly in your browser: https://huggingface.co/learn/deep-rl-course/unitbonus1/introduction
+  - A *longer tutorial* to understand how works ML-Agents: https://huggingface.co/learn/deep-rl-course/unit5/introduction
+
+  ### Resume the training
+  ```
+  mlagents-learn <your_configuration_file_path.yaml> --run-id=<run_id> --resume
+  ```
+
+  ### Watch your Agent play
+  You can watch your agent **playing directly in your browser** 
+
+  1. If the environment is part of ML-Agents official environments, go to https://huggingface.co/unity
+  2. Step 1: Find your model_id: {repo_id}
+  3. Step 2: Select your *.nn /*.onnx file
+  4. Click on Watch the agent play 👀 
+  """
+  
+    return model_card, metadata
+
+
+def _save_model_card(repo_dir: Path, generated_model_card: str, metadata):
+    """Save a model card to the repository.
+    :param repo_dir: repository directory
+    :param generated_model_card: model card generated by _generate_model_card() method
+    :param metadata: metadata
+    """
+    readme_path = repo_dir / "README.md"
+
+    with readme_path.open("w", encoding="utf-8") as f:
+        f.write(generated_model_card)
+
+    # Save our metrics to Readme metadata
+    metadata_save(readme_path, metadata)
+
+
+def package_to_hub(run_id,
+                   path_of_run_id,
+                   repo_id: str,
+                   commit_message: str,
+                   configfile_name
+                   ):
+    """
+    This method generates the model card and upload the run_id folder
+    with all his files into the Hub
+    :param run_id : name of the run
+    :param path_of_run_id: path of the run_id folder that contains the onnx model.
+    :param repo_id: id of the model repository from the Hugging Face Hub
+    :param commit_message: commit message
+    :param configfile_name: name of the yaml config file (by default configuration.yaml)
+    """
+    print(
+        f"This function will create a model card and upload your {run_id} "
+        f"into HuggingFace Hub. This is a work in progress: If you encounter a bug, "
+        f"please send open an issue")
+
+    _, repo_name = repo_id.split('/')
```

Comment:
```suggestion
    repo_name = repo_id.split('/')[-1]
```

(same as above - it's not wrong but I think this version is "safer")

Reply:
I think previous version was ok as we don't expect any legacy repos